Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs

```diff
@@ -226,20 +284,104 @@ public Task<TResult> InvokeWithParameterObjectAsync<TResult>(string targetName,
             await InvokeDidChangeConfigurationAsync(config);
         }
 
-        private IPythonLanguageClientContext CreateClientContext() {
+        private void OnWorkspaceFolderWatched() {
+            _workspaceFoldersSupported = true;
+            OnSolutionOpened();
+        }
+
+        private IPythonLanguageClientContext[] CreateClientContexts() {
             if (PythonWorkspaceContextProvider.Workspace != null) {
-                return new PythonLanguageClientContextWorkspace(PythonWorkspaceContextProvider.Workspace);
+                return new IPythonLanguageClientContext[] { new PythonLanguageClientContextWorkspace(PythonWorkspaceContextProvider.Workspace) };
             }
 
             if (ProjectContextProvider.ProjectNodes.MaybeEnumerate().Any()) {
-                // TODO: support multiple projects in solution.
-                var node = ProjectContextProvider.ProjectNodes.First();
-                if (node != null) {
-                    return new PythonLanguageClientContextProject(node);
+                var nodes = from n in ProjectContextProvider.ProjectNodes
+                            select new PythonLanguageClientContextProject(n);
+                return nodes.ToArray();
+            }
+
+            return new IPythonLanguageClientContext[] { new PythonLanguageClientContextGlobal(OptionsService) };
+        }
+
+        // This is all a hack until VSSDK LanguageServer can handle workspace folders
+        private StreamData OnSendToServer(StreamData data) {
+            var message = MessageParser.Deserialize(data);
+            if (message != null) {
+                try {
+                    // If this is the initialize method, add the workspace folders capability
+                    if (message.Value<string>("method") == "initialize") {
+                        if (message.TryGetValue("params", out JToken messageParams)) {
+                            var capabilities = messageParams["capabilities"];
+                            if (capabilities != null) {
+                                if (capabilities["workspace"] == null) {
+                                    capabilities["workspace"] = JToken.FromObject(new { });
+                                }
+                                capabilities["workspace"]["workspaceFolders"] = true;
```

Comment:
This is where the 'extra' capability is added. This allows pylance to support workspace folders